File: sp.iot.core/Model/Region.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;
using System.Linq;

namespace sp.iot.core
{
    public class Region : BaseItem
    {

        public Region()
        {
            Sections = new List<Section>();
            Rows = new List<RegionRow>();
        }

        [JsonPropertyName("name")]
        public string Name { get; set; }

        [JsonPropertyName("background-image")]
        public string BackgroundImage { get; set; }

        [JsonPropertyName("aspect-ratio")]
        public double AspectRatio { get; set; }

        [JsonPropertyName("type")]
        public RegionType Type { get; set; }

        [JsonPropertyName("sections")]
        public List<Section> Sections { get; set; }

        [JsonPropertyName("rows")]
        public List<RegionRow> Rows { get; set; }

        [JsonPropertyName("layout-rows-count")]
        public long RowsCount
        {
            get
            {
                if (Sections.Count > 0)
                    return Sections.Max(section => section.Row);
                else
                    return 0;
            }
        }

        [JsonPropertyName("layout-column-count")]
        public long ColumnCount
        {
            get
            {
                if (Sections.Count > 0)
                    return Sections.Max(section => section.Column);
                else
                    return 0;
               
            }
        }
    }
}

File: sp.client/lib/models/region.dart
import 'package:smart_places/models/row.dart';
import 'package:smart_places/models/section.dart';

class Region {
  String name;
  String type;
  String backgroundImage;
  double aspectRatio;
  List<Section> sections;
  List<Row> rows;
  int layoutRowsCount;
  int layoutColumnCount;
  String id;

  Region(
      {this.name,
      this.type,
      this.backgroundImage,
      this.aspectRatio,
      this.sections,
      this.rows,
      this.id});

  bool get hasVirtualLayoutInfo {
    return aspectRatio > 0 &&
        backgroundImage.isNotEmpty &&
        layoutColumnCount > 0 &&
        layoutRowsCount > 0;
  }

  int numberOfSectionHasGadgetsByType(String gadgetType) {
    int no = 0;
    sections.forEach((section) {
      if (section.numberOfGadgetsByType(gadgetType) > 0) {
        no++;
      }
    });
    return no;
  }

  int numberOfSectionHasGadgetsByGroup(String gadgetGroup) {
    int no = 0;
    sections.forEach((section) {
      if (section.numberOfGadgetsByGroup(gadgetGroup) > 0) {
        no++;
      }
    });
    return no;
  }

  Region copy() {
    return Region(
        name: this.name,
        type: this.type,
        backgroundImage: this.backgroundImage,
        aspectRatio: this.aspectRatio,
        id: this.id);
  }

  update(Region _region) {
    this.name = _region.name;
    this.type = _region.type;
    this.backgroundImage = _region.backgroundImage;
    this.aspectRatio = _region.aspectRatio;
  }

  Region.fromJson(Map<String, dynamic> json) {
    name = json['name'];
    type = json['type'];
    backgroundImage = json['background-image'];
    aspectRatio = json['aspect-ratio'].toDouble();
    if (json['sections'] != null) {
      sections = new List<Section>();
      json['sections'].forEach((v) {
        sections.add(new Section.fromJson(v));
      });
    }
    if (json['rows'] != null) {
      rows = new List<Row>();
      json['rows'].forEach((v) {
        rows.add(new Row.fromJson(v));
      });
    }
    layoutRowsCount = json['layout-rows-count'];
    layoutColumnCount = json['layout-column-count'];
    id = json['id'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['"name"'] = '"' + this.name + '"';
    data['"type"'] = '"' +  this.type + '"';
    data['"background-image"'] = '"' + this.backgroundImage + '"';
    data['"aspect-ratio"'] = this.aspectRatio;

    if (this.sections != null) {
      data['"sections"'] = this.sections.map((v) => v.toJson()).toList();
    }
    if (this.rows != null) {
      data['"rows"'] = this.rows.map((v) => v.toJson()).toList();
    }
    data['"id"'] = '"' + this.id + '"';
    return data;
  }
}
